labshare-client/src/backend/endpoints/admin.ts

```diff
@@ -0,0 +1,289 @@
+import express from "express"
+import utils from '../utils'
+import { UNAUTHORIZED, INTERNAL_SERVER_ERROR, BAD_REQUEST, NOT_FOUND } from 'http-status-codes'
+import { UserAdmin, UserCommon, ActivationToken } from '../database/database'
+import argon from 'argon2'
+import JsonSchema, { schemas } from "../jsonSchemas/JsonSchema";
+import { CONF } from '../options'
+import jsonwebtoken from "jsonwebtoken"
+import { registration_admin } from '../jsonSchemas/registration_admin'
+import { Validator } from 'jsonschema'
+import { authMiddleware } from '../middlewares/auth'
+import { sendActivationMail } from '../mail/mailer'
+import { getLangID } from './language'
+import { v4 } from 'uuid'
+
+
+export class AdminEndpoint {
+    router: express.IRouter
+
+    constructor(router: express.IRouter) {
+        this.router = router
+        this.setupRoutes()
+    }
+
+    setupRoutes() {
+        this.router.post('/login', this.login)
+        
+        this.router.use(this.adminAuthMiddleware)
+        this.router.post('/create', this.createAdmin)
+        this.router.delete('/delete', this.deleteAdmin)
+        this.router.get("/users", this.getProfiles)
+        this.router.post('/user/activate/:id', this.activateUser)
+        this.router.post('/user/disable/:id', this.disableUser)
+        this.router.post('/user/enable/:id', this.enableUser)
+        this.router.post('/user/delete/:id', this.deleteUser)
+        this.router.post('/user/resendActivationMail/:id', this.resendActivationMail)
+
+        this.router.use(this.blackhole)
+    }
+
+    async adminAuthMiddleware(req: express.Request, res: express.Response, next: express.NextFunction) {
+        authMiddleware(req, res, function() {
+            let token = utils.getUnverifiedDecodedJWT(req)
+            if (token.role !== 'admin' && token.role !== 'superAdmin') {
```

Comment:
These roles should be compared against `USER_ROLE.{role}`

Reply:
Done